src/store/getters.ts

```diff
@@ -283,5 +285,36 @@ export default {
   /** The entity TodoList resources. */
   getTodoListResource (state: StateIF): TodoListResourceIF {
     return state.configObject?.todoList
+  },
+
+  /** A formatted concatenation of the name and the effectiveDate of the filing  */
+  getReasonText (state: StateIF, getters): string {
+    const filingType: FilingTypes = state.stateFiling.header.name
+    // create reason text to display in the info header
+    let name: string
+    let date: string
+
+    if (filingType === FilingTypes.DISSOLUTION) {
+      name = EnumUtilities.dissolutionTypeToName(getters.isFirm,
+        (state.stateFiling?.dissolution?.dissolutionType as DissolutionTypes) ||
+        DissolutionTypes.UNKNOWN
+      )
+      const dissolutionDate = DateUtilities.yyyyMmDdToDate(state.stateFiling.dissolution?.dissolutionDate)
+      if (!dissolutionDate) throw new Error('Invalid dissolution date')
+      date = DateUtilities.dateToPacificDate(dissolutionDate, true)
+    } else {
+      name = EnumUtilities.filingTypeToName(filingType)
+      const effectiveDate = this.apiToDate(state.stateFiling.header?.effectiveDate)
+      if (!effectiveDate) throw new Error('Invalid effective date')
+      date = DateUtilities.dateToPacificDateTime(effectiveDate)
+    }
+
+    const enDash = '–' // ALT + 0150
+    return `${name} ${enDash} ${date}`
+  },
+
+  /** */
+  isEntityInLimitedRestoration (state: StateIF): boolean {
+    return (state.stateFiling?.business?.state === 'ACTIVE' && state.stateFiling.hasOwnProperty('restoration'))
```

Comment:
It may be the purview of a new ticket, but I'm thinking we should ideally/eventually keep "pure getters" (which return store data) with these "computed getters". Thoughts?

Reply:
Hmm, let me think about that.  My gut reaction is that I don't see the benefit of segregating getters in this way.  For example, if I'm calling a getter called `isDisplaySomeRandomComponent()`, I shouldn't have wonder if it's a pure getter or a computed getter.  How it works should be entirely hidden from me.

Reply:
Would the "store" just be one thing that we import/use getters from, regardless of where they're declared?

Reply:
I like your idea of having a number of stores organized by the resource.  For example, filingHistory's getters, mutators and actions would be segregated from the ToDo's getters, setters and mutators.

Reply:
I thought that was your idea? :)

Todo List vs FHL might not be a good example since they have few setters/actions, but yes to the general idea.

I'm still thinking along the lines of "pure" getters/actions (to access state) vs "complex/computed/rich" getters -- which might not need to be in the store at all? Dunno, depends on what they need to do, and I'm keeping my eyes open to what you're updating and proposing.

Reply:
Excellent, I'll continue to make small changes so we can discuss tangible examples rather than hypotheticals.  Again, I SO appreciate you being open to these discussions.